daemon/persistence/sampletask_dao.go

```diff
@@ -2,52 +2,61 @@ package persistence
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"github.com/singerdmx/BulletJournal/daemon/config"
 	"gorm.io/gorm"
 )
 
 type SampleTaskDao struct {
 	Ctx context.Context
-	db *gorm.DB
+	Db  *gorm.DB
+}
+
+func InitializeSampleTaskDao(config *config.Config, ctx context.Context) *SampleTaskDao {
+	return &SampleTaskDao{
+		Ctx: ctx,
+		Db: NewDB(config),
+	}
 }
 
 func NewSampleTaskDao() (*SampleTaskDao, error) {
 
 	sampleTaskDao := SampleTaskDao{
-		db: DB,
+		Db: DB,
 	}
 	return &sampleTaskDao, nil
 }
 
 func (s *SampleTaskDao) Upsert(t *SampleTask) {
-	//prevReport := SampleTask{}
-	//err := s.db.Where("uid = ?", t.Uid).Last(&prevReport).Error
-	//if errors.Is(err, gorm.ErrRecordNotFound) {
-	//	err :=s.db.Create(&t).Error
-	//	if err != nil {
-	//		fmt.Println(err, t.ID)
-	//	}
-	//} else {
-	//	s.db.First(&t)
-	//	s.db.Save(&t)
-	//}
+	prevReport := SampleTask{}
+	err := s.Db.Where("uid = ?", t.Uid).Last(&prevReport).Error
+	if errors.Is(err, gorm.ErrRecordNotFound) {
+		err :=s.Db.Create(&t).Error
+		if err != nil {
+			fmt.Println(err, t.ID)
+		}
+	} else {
+		s.Db.Save(&t)
```

Comment:
only update column "content", not the whole thing

Reply:
If you can load it from DB, update "content" field, and save the object loaded from DB, not the one passed into the function

Reply:
I see I will adjust this sometime tomorrow

Reply:
Add duedate, duetime change for the case of IPO deadline changed